File: ziteng/Nth to Last Node in List.java
/**
 * Definition for ListNode.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int val) {
 *         this.val = val;
 *         this.next = null;
 *     }
 * }
 */ 
public class Solution {
    /**
     * @param head: The first node of linked list.
     * @param n: An integer.
     * @return: Nth to last node of a singly linked list. 
     */
    ListNode nthToLast(ListNode head, int n) {
        ListNode p = head;
        int len = 0;
        while(p!=null) {
            p = p.next;
            len++;
        }
        p = head;
        for(int i = 0; i < len - n; i++) {
            p = p.next;
        }
        return p;
    }
}

File: Sen/Nth_to_Last_Node_in_List.py
"""
Definition of ListNode
class ListNode(object):

    def __init__(self, val, next=None):
        self.val = val
        self.next = next
"""
class Solution:
    """
    @param head: The first node of linked list.
    @param n: An integer.
    @return: Nth to last node of a singly linked list. 
    """
    def nthToLast(self, head, n):
        # write your code here
        start = ListNode(0)
        slow = start
        fast = start 
        slow.next = head
        for i in range(1, n + 2):
            fast = fast.next
        while fast:
            fast = fast.next
            slow = slow.next
        return slow.next
